uaclient/entitlements/esm.py

```diff
@@ -12,3 +12,15 @@ class ESMEntitlement(repo.RepoEntitlement):
     repo_key_file = "ubuntu-advantage-esm-infra-trusty.gpg"
     repo_pin_priority = "never"
     disable_apt_auth_only = True  # Only remove apt auth files when disabling
+
+
+class ESMLegacyEntitlement(repo.RepoEntitlement):
+
+    help_doc_url = "https://ubuntu.com/esm"
+    name = "esm-infra-legacy"
+    title = "Ubuntu Pro: ESM Infra (Legacy)"
+    origin = "UbuntuESM"
+    description = "Expanded Security Maintenance for Infrastructure on Legacy Instances"  # noqa
+    repo_url = "https://esm.ubuntu.com/infra-legacy/"
+    repo_key_file = "ubuntu-advantage-esm-infra-legacy-trusty.gpg"
```

Comment:
Do we also need to handle this new key file in postinst and postrm?

Reply:
Indeed, I have updated `postrm` for the purge scenario.

For postinst, it is not necessary as we don't want to configure "advertisment` for `esm-infra-legacy` as we do for `esm-infra`. So I am not touching that file